include/cooking/cookingQuerys/ratingSystem.php

```diff
@@ -0,0 +1,136 @@
+        ")->fetchALL();
+}
+function listOfAllcookingTags(){
+    $tagTypes=getAllCookingTags();
+    $tagArray = array();
+    $returnTags = "";
+    foreach ($tagTypes as $tagsOrganized){
+        if(!@$tagArray[$tagsOrganized['recipeIngredientName']]){
+            $tagArray[$tagsOrganized['recipeIngredientName']] = true;
+        }
+        $returnTags .= "<br/><div class='resume'> ".$tagsOrganized['recipeIngredientName']." </div>";
+    }
+    return $returnTags;
+}
+function addNewTagToExistingRecipe($tagName, $recipeId){
+    dbQuery("INSERT INTO recipe_identifier_linked(recipeIngredientName, recipeId)
+        VALUES (:recipeIngredientName, :recipeId)", array('recipeIngredientName'=>$tagName, 'recipeId'=>$recipeId));
+            //header('Location:cookingWebsite/yourCookbook/yourRecipes.php');
+}
+function deleteTagFromExistingRecipes($tagName, $postId){
+    dbQuery("DELETE FROM recipe_identifier_linked WHERE recipeIngredientName=:recipeIngredientName and recipeId=:recipeId",
+        array('recipeIngredientName'=>$tagName, 'recipeId'=>$postId));
+        //header('Location:cookingWebsite/yourCookbook/yourRecipes.php');
+}
+function addNewCookingTagQuery ($tagName){
+        dbQuery("INSERT INTO recipeTagIds (recipeIngredientName)
+            VALUES (:recipeIngredientName)", array('recipeIngredientName'=>$tagName));
+            header('Location:/cookingWebsite/yourCookbook/addTags.php');
+}
+function deleteNewCookingTagQuery ($tagName){
+        dbQuery("DELETE FROM recipeTagIds WHERE recipeIngredientName=:recipeIngredientName", array('recipeIngredientName'=>$tagName));
+        dbQuery("DELETE FROM recipe_identifier_linked WHERE recipeIngredientName= :recipeIngredientName", array('recipeIngredientName'=>$tagName));
+        return header('Location:/cookingWebsite/yourCookbook/addTags.php');
+}
+
+// ********ALL FUNCTIONS FOR RATINGS BELOW*********
+
+function insertRatingIntoTable($loginId, $recipeId, $rating){
+    dbQuery("INSERT INTO ratings(loginId, recipeId, rating)
+            VALUES(:loginId, :recipeId, :rating)",
+            array('loginId'=>$loginId, 'recipeId'=>$recipeId, 'rating'=>$rating));
+}
+function algorithm(){
+    $total = 0;
+    $loginQuery = dbQuery("SELECT * FROM cookingLogin")->fetchALL();
+    $loginArray = array();
+        //$returnRecipeTags = "";
+        foreach ($loginQuery as $logins){
+            if(!@$loginArray[$logins['loginId']]){
+                $loginArray[$logins['loginId']] = true;
+            }
+                $loginId = $logins['loginId'];
+
+
+    $recipeQuery = dbQuery("SELECT * FROM recipes")->fetchALL();
+    $recipeArray = array();
+        //$returnRecipeTags = "";
+        foreach ($recipeQuery as $recipes){
+            if(!@$recipeArray[$recipes['recipeId']]){
+                $recipeArray[$recipes['recipeId']] = true;
+            }
+                $recipeId = $recipes['recipeId'];
+                $tagQuery = dbQuery("SELECT * FROM recipe_identifier_linked
```

Comment:
I think this could be moved outside the user loop also. Basically, do a bit of work at the beginning to pre-compute these arrays so that you can reduce the number of queries you're running.